ocaml/typing/typeopt.ml

```diff
@@ -234,12 +234,16 @@ let bigarray_type_kind_and_layout env typ =
       (Pbigarray_unknown, Pbigarray_unknown_layout)
 
 let value_kind_of_value_jkind jkind =
-  match Jkind.get_default_value jkind with
-  | Value -> Pgenval
-  | Immediate -> Pintval
-  | Immediate64 ->
+  let const_jkind = Jkind.default_to_value_and_get jkind in
+  let externality_upper_bound =
+    Jkind.Const.get_externality_upper_bound const_jkind
+  in
+  (* CR: assert the sort is a value *)
```

Comment:
Yes, indeed. But why not just do this? I don't think this needs to be delayed.

Reply:
I meant to come back to it and then forgot to. Fixed.